### PR TITLE
Feature/assignment geostore

### PR DESCRIPTION
**Pull request by edward3sc**
Add an alertType field into the assignment.location object.

Add a geostore field into the assignment document that will hold a geostoreId. 

Body in POST /assignments now requires a geojson in the geostore field which gets saved as a geostore using GeostoreService and the geostore id saved in the assignment.geostore field.

Responses to GET requests now fetch the geostore using the id from the GeostoreService.getGeostore function and returning the assignment with the fetched geostore object as the assignment.attributes.geostore field

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 FROM node:14
 
 # Create app directory
-WORKDIR /usr/src/app
```
Comment (correcthorsebatterystaple) [inline Dockerfile:5]: Why are we changing the Dockerfile?

File: src/assignments/assignments.controller.ts [hunk 2]
```diff
@@ -36,7 +36,13 @@ export class AssignmentsController {
       request.user,
     );
 
-    return { data: serializeAssignments(createdAssignment) };
+    const [assignmentResponse] =
+      await this.assignmentsService.buildAssignmentResponse(
```
Comment (correcthorsebatterystaple) [inline src/assignments/assignments.controller.ts]: I would steer away from putting response builder logic in the service, would rather inject the geostore service into the controller then fetch the data and pass it off to a response builder method or have a response builder method in the controller.

File: src/assignments/assignments.service.ts [hunk 3]
```diff
@@ -61,9 +64,18 @@ export class AssignmentsService {
           .join('')
       : 'null';
 
+    // create geostore
+    let geostore: IGeostore = null;
+    if (assignment.geostore)
+      geostore = await this.geostoreService.createGeostore(
+        assignment.geostore,
+        user.token,
+      );
```
Comment (correcthorsebatterystaple) [inline src/assignments/assignments.service.ts]:
```suggestion
    let geostore: IGeostore = assignment.geostore ?
        await this.geostoreService.createGeostore(
          assignment.geostore,
          user.token,
        ) : null;
```

File: src/assignments/assignments.service.ts [hunk 4]
```diff
@@ -61,9 +64,18 @@ export class AssignmentsService {
...
+
     const newAssignment = {
       ...assignment,
       createdBy: user.id,
+      geostore: geostore.id ? geostore.id : null,
```
Comment (correcthorsebatterystaple) [inline src/assignments/assignments.service.ts:78]:
```suggestion
      geostore: geostore.id ?? null,
```

File: src/assignments/tests/assignments.e2e.spec.ts [hunk 5]
```diff
@@ -524,6 +582,8 @@ describe('Assignments', () => {
         .set('Authorization', 'USER')
         .expect(200);
 
+      console.log('*****', response.body.data[1].attributes.geostore);
```
Comment (correcthorsebatterystaple) [inline src/assignments/tests/assignments.e2e.spec.ts]:
```suggestion
```

File: tmp/0efc47b985ea42acfded0e7e05e7236d [hunk 6]
```diff
@@ -0,0 +1 @@
+TestFileContent
```
Comment (correcthorsebatterystaple) [inline tmp/0efc47b985ea42acfded0e7e05e7236d]: Remove these temp files

**Review by correcthorsebatterystaple — CHANGES_REQUESTED** [review]

File: src/assignments/assignments.service.ts [hunk 7]
```diff
@@ -153,4 +165,21 @@ export class AssignmentsService {
     });
     return null;
   }
+
+  async buildAssignmentResponse(
+    assignments: AssignmentDocument[],
```
Comment (correcthorsebatterystaple) [inline src/assignments/assignments.service.ts]: Can we overload the function so it takes both the array and the single version?

**Review by correcthorsebatterystaple — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (edward3sc) [reply Dockerfile:5]: Trying to get eslint to work! Seems to work fine with the changes. If mine clash with yours then I'll revert back

File: src/assignments/assignments.controller.ts (hunk 2 above)
Comment (edward3sc) [reply src/assignments/assignments.controller.ts]: Done

File: src/assignments/assignments.service.ts (hunk 7 above)
Comment (edward3sc) [reply src/assignments/assignments.service.ts]: We can, but I think it's easier to always receive and respond with an array and none of the rest of the code needs to change

File: Dockerfile (hunk 1 above)
Comment (correcthorsebatterystaple) [reply Dockerfile:5]: image
This is what I went for updated the github workflow

Comment (edward3sc) [reply Dockerfile:5]: The Dockerfile was a bit of a piece-together in the first place. I think these changes make it a bit more robust

**Review by correcthorsebatterystaple — APPROVED** [review]

**Pull request merged**